app/errors/errors_test.go

```diff
@@ -0,0 +1,145 @@
+package errors_test
+
+import (
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/errors"
+	"github.com/celestiaorg/celestia-app/app"
+	apperr "github.com/celestiaorg/celestia-app/app/errors"
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+	testutil "github.com/celestiaorg/celestia-app/test/util"
+	blob "github.com/celestiaorg/celestia-app/x/blob/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// This will detect any changes to the DeductFeeDecorator which may cause a
+// different error message that does not match the regexp.
+func TestInsufficientMinGasPriceIntegration(t *testing.T) {
+	var (
+		gasLimit  uint64 = 1_000_000
+		feeAmount int64  = 10
+		gasPrice         = float64(feeAmount) / float64(gasLimit)
+	)
+	account := "test"
+	testApp, kr := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams(), account)
+	minGasPrice, err := sdk.ParseDecCoins(fmt.Sprintf("%v%s", appconsts.DefaultMinGasPrice, app.BondDenom))
+	require.NoError(t, err)
+	ctx := testApp.NewContext(true, tmproto.Header{}).WithMinGasPrices(minGasPrice)
+	signer := blob.NewKeyringSigner(kr, account, testutil.ChainID)
+	builder := signer.NewTxBuilder()
+	builder.SetGasLimit(1_000_000)
+	builder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(10))))
```

Comment:
[question] did these lines intend to use the variables defined above? I.e.

```suggestion
	builder.SetGasLimit(gasLimit)
	builder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(feeAmount))))
```

Reply:
Yeah good catch